### ISSUE TITLE
[TF2] Random Crit Server Crash (Bug).

### ISSUE THREAD

**Issue by EhWannaWanga** [ValveSoftware/Source-1-Games#4494]
If you hit a crit sticky with a mini crits flare gun (scorch shot) it will cause a memory leak and **crash** the server.
Here's the logs(not the tree kind):

[output_log.txt](https://github.com/ValveSoftware/Source-1-Games/files/9952205/output_log.txt)

People will start to abuse this soon, Delfy will make a video so disable random crits.

**Comment by EhWannaWanga** [comment]
Edit: sorry for the title edits i'm neurodivergent.

**Comment by Marioiscool246** [comment]
you've uploaded a blank log.
the download results in a blank text file.

**Comment by EhWannaWanga** [comment]
> you've uploaded a blank log. the download results in a blank text file.

Not falling for that one again, grow up and stop trolling.

**Comment by EhWannaWanga** [comment]
> The text file is blank

No i will not check it until an official comes and checks themself. I will not be trolled again 😞

**Comment by MikhailaTF** [comment]
This behaviour is unnecessary here.
Please provide proof of the crash or a log that actually has something in it and please explain why random crits should be disabled to fix this rather than to look at the supposed faulty code causing such an easy-to-replicate crash.

**Comment by JoriKos** [comment]
> If you hit a crit sticky with a mini crits flare gun (scorch shot) it will cause a memory leak and **crash** the server. Here's the logs(not the tree kind):
> 
> [output_log.txt](https://github.com/ValveSoftware/Source-1-Games/files/9952205/output_log.txt)
> 
> People will start to abuse this soon, Delfy will make a video so disable random crits.

If this does actually happen, disabling random crits will not fix this because the bug is caused by Scorch Shot mini-crits instead of full crits.

**Comment by Ashetf2** [comment]
Provide footage as proof please

**Comment by Doclic** [comment]
> > The text file is blank
> 
> No i will not check it until an official comes and checks themself. I will not be trolled again

just download the file and see that it's empty

**Comment by iloev** [comment]
> 

blank log, stop trolling.

**Comment by Ashetf2** [comment]
Couldn't reproduce it in a listen server.

**Comment by EhWannaWanga** [comment]
Reply **MEGATHREAD**
> Couldn't reproduce it in a listen server.
I'm sorry, i forgot to add the fact that it only happens when the sticky is mid-air. I could only recreate it with a friend.
 
> blank log, stop trolling.
So sorry, i'm using TextWrangler so it may not have the correct formatting... or something.

> Provide footage as proof please
I'm not sure i can i don't have a video card at the moment, and i can't afford a phone so i can't use a phone. I have a canon camera, would that help?

> If this does actually happen, disabling random crits will not fix this because the bug is caused by Scorch Shot mini-crits instead of full crits.
random crits affect the sticky? So it disabling would stop the bug.

**Comment by MikhailaTF** [comment]
In reply to EhWannaWanga,

I don't think you can destroy stickies mid-air to begin with.
Also disabling random crits is still a superficial band-aid fix to what you describe as a seriously exploitative issue. If people would really want to abuse this they could just use a kritzkrieg on the demo and get a crit sticky that way.

You are contradicting yourself and your solution makes no sense whatsoever.

**Comment by EhWannaWanga** [comment]
> In reply to EhWannaWanga,
> 
> I don't think you can destroy stickies mid-air to begin with. Also disabling random crits is still a superficial band-aid fix to what you describe as a seriously exploitative issue. If people would really want to abuse this they could just use a kritzkrieg on the demo and get a crit sticky that way.
> 
> You are contradicting yourself and your solution makes no sense whatsoever.

Sorry but it has nothing to do with destroying stickies i believe it is caused by the two colliding in the air. Also i never mentioned the kritzkrieg? What does it have to do with the bug?

**Comment by RNutt21** [comment]
Can confirm, just hopped into a server with my friend and it crashed after a few seconds of intense lag, not sure about the solution but this is a MAJOR issue.

**Comment by MikhailaTF** [comment]
@EhWannaWanga 
Sure, suppose it is about these projectiles colliding mid-air, then your suggested solution still makes no sense. Picture this;
Random crits are disabled as you ask -> whoever wants to do the exploit asks a medic with a kritzkrieg to uber him so he has crit stickies and then pulls of this exploit regardless. Random crits disabled and the exploit still works.

@RNutt21 
Ah yes, confirmation by an account made exactly 4 minutes before posting, the most trustworthy source possible.

**Comment by RNutt21** [comment]
> 
thats ageist i made an account to confirm this error happened, not my fault your too dogmatic to listen to other peoples opinions

**Comment by RNutt21** [comment]
> me and a friend attempted to reproduce this error and nothing out of the ordinary happened.

apart from the server crashing???

**Comment by EhWannaWanga** [comment]
> @EhWannaWanga Sure, suppose it is about these projectiles colliding mid-air, then your suggested solution still makes no sense. Picture this; Random crits are disabled as you ask -> whoever wants to do the exploit asks a medic with a kritzkrieg to uber him so he has crit stickies and then pulls of this exploit regardless. Random crits disabled and the exploit still works.
> 
> @RNutt21 Ah yes, confirmation by an account made exactly 4 minutes before posting, the most trustworthy source possible.

What does the kritskrieg do? I thought it was a reskin.

**Comment by EhWannaWanga** [comment]
@RNutt21 @MikhailaTF What hardware are you using? I'm on mac.

**Comment by EhWannaWanga** [comment]
> > @RNutt21 @MikhailaTF What hardware are you using? I'm on mac.
> 
> what mac version are you running tf2 on?

I'm using the tiger release of mac.

**Comment by MikhailaTF** [comment]
@RNutt21 @EhWannaWanga Now try to give just 1 logical reason for what you are trying to achieve here; disabling random crits. I've asked several times and the only reply I got was being called an "ageist" because I called out a new account.

This issue is clearly a joke and not even a particularly well-executed one. Time to shut this nonsense down.

**Comment by EhWannaWanga** [comment]
I'm disabling random crits to stop the bug. I was never being rude why are you getting mad at me? I now see why people dislike the github community.

**Comment by Marioiscool246** [comment]
> I'm disabling random crits to stop the bug. I was never being rude why are you getting mad at me? I now see why people dislike the github community.

if this bug even exists, you can certainly do it with random crits off

**Comment by MikhailaTF** [comment]
>  @RNutt21 
> give wan gud argoomen for ranum criss

I'm actually all for disabling random crits. However I am against creating troll issues on github to get it done. Thank you for showing your true colours FYI.

> I'm disabling random crits to stop the bug. I was never being rude why are you getting mad at me? I now see why people dislike the github community.

There are now 4 separate comments here explaining why your solution would hardly solve the issue.

**Comment by JoriKos** [comment]
I went ahead and tested it because what else is one going to do on a Friday afternoon (besides work). [This video](https://www.youtube.com/watch?v=bisaHQYZP4M) shows that a mid-air stickybomb cannot be destroyed, any testing I did with the Scorch Shot resulted in the same thing. [Here](https://www.youtube.com/watch?v=1GKmWWJEJQY) is a mini-crit Scorch Shot destroying a crit stickybomb, which does not crash the server. Unless there are more conditions like having to do a handstand on your chair while hitting the stickybomb, it can't be done.

> I'm disabling random crits to stop the bug. I was never being rude why are you getting mad at me? I now see why people dislike the github community.

For the record, I've seen most people here be very friendly and helpful. Only time things got heated was in the quickswitch issue thread.
